### PR TITLE
feat: add button to toggle dark/light mode

### PR DESCRIPTION
**Pull request by Alkindi42**
Hello,

I have added a toggle button to switch between light/dark mode (#306).

I also added a dark background color for the dark mode. I just saw it's a duplicate with #307, I can rollback this part if needed.

Screenshot_2021-03-03-182230
Screenshot_2021-03-03-182214

### CODE REVIEW

**Comment by Misterio77** [comment]
Just wanted to let you know, it seems the checkbox is not appearing for me? This also seem to have broken auto-theming based on `prefers-color-scheme`:
image

**Comment by g33kex** [comment]
I like the idea of being able to toggle dark mode (diagrams are not very readable in dark mode, but perhaps this is another bug), but I'm not sure a checkbox is the best idea. 
I usually use the print function of Firefox to export the preview page to pdf to quickly share my notes, and that would add the button to the preview. Maybe a shortcut would be better? What do you think?

**Comment by Misterio77** [comment]
I was thinking about `:MarkdownPreview [light/dark]`, with the default being the sheet that sets the colors based on `prefers-color-scheme`. I would implement it, but i'm not sure how to do it.

**Comment by Alkindi42** [comment]
@Misterio77  I have rebuilt the app, it should work for you. By default, we apply user preference based on `prefers-color-scheme`. The user can override this choice by the checkbox button.

@g33kex I think you don't use the right tool. When you share your note from Firefox you also have local address at the top non?
Maybe you can use a tool like [Pandoc](https://pandoc.org/)? (for the vim integration you case use [vim-pandoc](https://github.com/vim-pandoc/vim-pandoc)) to convert your document in the PDF format.

In the case you really want to remove the button, we have several solutions:
- We can click on the header or add a keyboard shortcut to change the theme. I'm not really fan of this solution because the user's experience is bad.. (information hidden somewhere in the documentation)
- Implement the @Misterio77's solution even if for me the solution it's not perfect. Being able to switch the theme on the fly from the rendering seems important to me. What do you think?
- Instead of using a "MarkdownPreview" command with an optional argument, we can add a new configuration variable to override the theme (`let g:mkdp_theme = '"dark"). However we have the same limitation as the above solution in addition to being able to change the theme "less quickly".

Thank you for you feedback!

**Comment by g33kex** [comment]
@Alkindi42 Thank you for your answer! Pandoc unfortunately uses a somewhat different Markdown flavor, and doesn't support out of the box everything supported here (diagrams, ...). I'd really like to have the same rendering on the export and on the preview for consistency. 
Feature request #87 asks for a pdf export, so if it is implemented, the button would have to be removed somehow before export?

I agree it should be changed on the fly, but I don't see why it couldn't be changed from vim. Vim needs to be opened anyway, so if a Vim commands allows to change the theme on the fly, one could easily bind it to a shortcut. I don't how your use case, but I personally never even focus the browser window.

**Comment by iamcco** [comment]
Thanks, please rebase master, seems conflict with previous pr.

**Comment by iamcco** [comment]
> I like the idea of being able to toggle dark mode (diagrams are not very readable in dark mode, but perhaps this is another bug), but I'm not sure a checkbox is the best idea.
> I usually use the print function of Firefox to export the preview page to pdf to quickly share my notes, and that would add the button to the preview. Maybe a shortcut would be better? What do you think?

Another way is that the checkbox only visible when mouse hover on it.

File: app/_static/page.css [hunk 1]
```diff
@@ -14,7 +18,25 @@
 }
```
Comment (iamcco) [inline app/_static/page.css:18]:
```css
@media (prefers-color-scheme: dark) {
  :root {
    --foreground-color: #c9d1d9;
    --background-color: #0d1117;
    --border-color: #30363d;
    --body-color: #000000;
  }
}
```

Is this still necessary?

Comment (Alkindi42) [reply app/_static/page.css]: No I removed it

**Comment by Alkindi42** [comment]
> I agree it should be changed on the fly, but I don't see why it couldn't be changed from vim. Vim needs to be opened anyway, so if a Vim commands allows to change the theme on the fly, one could easily bind it to a shortcut.

You are right, I misunderstood ;)

> Thanks, please rebase master, seems conflict with previous pr.

It's done.

> I don't how your use case, but I personally never even focus the browser window.

To be honest, I don't use this plugin a lot. I was just frustrated that I couldn't see anything on my Plantuml diagram and I had some time ;)

I have implemented the solution with the mouse over. When the cursor is in the header, the checkbox is visible. I put the information on the hidden button in the README.

Is it ok for everyone?

**Comment by nyngwang** [comment]
@Alkindi42 Could you implement the option `let g:mkdp_theme` to set the default theme? There are two new issues about this, apparently not everyone like this idea, and it's a little annoying to have default set to dark.

**Pull request merged**